client/finality-grandpa/src/lib.rs

```diff
@@ -858,6 +877,10 @@ where
 					last_finalized,
 				);
 
+				// Repoint shared_voter_state so that the RPC endpoint can query the state
+				let mut shared_voter_state = self.shared_voter_state.write();
```

Comment:
Just to be safe, could you replace this `.write()` with `.try_write_for(Duration::from_secs(1))`? Just to be safe, in case there's a deadlock we won't block the voter.

Reply:
Yes that makes sense!